repo/repo.go

```diff
@@ -89,17 +99,54 @@ func ListRepositories(workspace *rule.File) (repos []*rule.Rule, repoFileMap map
 			if err != nil {
 				return nil, nil, err
 			}
+			directiveRepoMap, err = parseRepositoryDirectives(macroFile.Directives)
+			if err != nil {
+				return nil, nil, err
+			}
+			for _, repo := range directiveRepoMap {
+				repos = append(repos, repo)
+				repoFileMap[repo.Name()] = macroFile
+			}
 			for _, repo := range macroFile.Rules {
 				if name := repo.Name(); name != "" {
-					repos = append(repos, repo)
-					repoFileMap[name] = macroFile
+					if _, ok := directiveRepoMap[name]; !ok {
+						repos = append(repos, repo)
+						repoFileMap[name] = macroFile
+					}
 				}
 			}
 		}
 	}
 	return repos, repoFileMap, nil
 }
 
+func parseRepositoryDirectives(directives []rule.Directive) (repoMap map[string]*rule.Rule, err error) {
+	repoMap = make(map[string]*rule.Rule)
+	for _, d := range directives {
+		switch d.Key {
+		case "repository":
+			vals := strings.Split(d.Value, " ")
+			if len(vals) < 2 {
+				return nil, fmt.Errorf("Failure parsing repository: %s, you must define at least a rule kind and name attribute", d.Value)
```

Comment:
Nit: errors should start with lowercase letters (golint nags me about this) and should not address the user directly. 

Also, no need to say the name attribute is missing because we check for that below. So maybe if `len(vals) == 0`, then `parsing repository directive: expected repository kind and attributes`.